### ISSUE TITLE
Add defaults to the YeelightBulb class.

### ISSUE THREAD

**Issue by skorokithakis** [HydrelioxGitHub/PyYeelight#1]
It would be nice if the defaults could be overridden when instantiating `YeelightBulb`. For example:

~~~
pyyeelight.YeelightBulb(ip="192.168.0.1", effect=effect=EFFECT_SMOOTH, transition_time=300)
~~~

Then, all subsequent commands would use these defaults.

This can be achieved easily with two properties:

~~~
YeelightBulb.effect = EFFECT_SUDDEN
YeelightBulb.transition_time = MIN_TRANSITION_TIME
~~~

Then, each command would have something like:

~~~
def set_hsv_color(self, hue, saturation, effect=None, transition_time=None):
    # Code...
    schema({'hue': hue, 'saturation': saturation, 'effect': effect or self.effect, 'transition_time': transition_time or self.transition_time})
~~~

**Comment by HydrelioxGitHub** [comment]
Good idea, I 'm a little busy for the next week. But I will try to implement this when I will have some free time.

**Comment by skorokithakis** [comment]
I might issue a few pull request, I need this for the cli I'm writing. Will you at least be available to merge them and push to pypi?

**Comment by HydrelioxGitHub** [comment]
As I write on Twitter, the future development of this library will be done to handle rgb bulb and White bulb as well. 
I think there will be a generic bulb class and some rgb class and white class with specific method and heritage from the generic class. 

I will welcome your PR when this structure will be ok. I also need to write some unit tests if the library become collaborative to be sure that modification will not break something.  I hope to achieve this early in december. 

Sorry for being so long !

**Comment by skorokithakis** [comment]
Hmm, will this structure change affect anything in my PR? I'm getting my utility ready now, so I would like this functionality to be in the first version. I can add it to pyyeelight directly, or I can add it to my utility, but if I do the latter I doubt I'll have time to do the former later on...

I don't really mind either way, so it's up to you, but I'd be happy to submit a PR for this now if you want. Otherwise, we can revisit it later. Thanks for your efforts!

**Comment by HydrelioxGitHub** [comment]
I think your idea is an enhancement for this library so please PR ! 

I will merge it  and push it to pypi at the end of the next week. When I will be back at home from my trip.
